code/parachain/frame/staking-rewards/src/lib.rs

```diff
@@ -1641,21 +1641,15 @@ fn update_rewards_pool<T: Config>(
 
 		for (asset_id, update) in reward_updates {
 			let reward = pool.rewards.get_mut(&asset_id).ok_or(Error::<T>::RewardAssetNotFound)?;
-			match do_reward_accumulation::<T>(asset_id, reward, &pool_account, now_seconds) {
+			match do_reward_accumulation::<T>(pool_id, asset_id, reward, &pool_account, now_seconds)
+			{
 				Ok(()) => {},
 				Err(RewardAccumulationCalculationError::BackToTheFuture) =>
 					return Err(Error::<T>::BackToTheFuture.into()),
-				Err(RewardAccumulationCalculationError::MaxRewardsAccumulated) => {
-					Pallet::<T>::deposit_event(Event::<T>::MaxRewardsAccumulated {
-						pool_id,
-						asset_id,
-					});
-					continue
-				},
-				Err(RewardAccumulationCalculationError::MaxRewardsAccumulatedPreviously) =>
-					continue,
 				Err(RewardAccumulationCalculationError::RewardsPotEmpty) =>
 					return Err(Error::<T>::RewardsPotEmpty.into()),
+				Err(RewardAccumulationCalculationError::ArithmeticError) =>
+					return Err(Error::<T>::ArithmeticError.into()),
```

Comment:
is this error still required?

Reply:
It is, given the current use of
 ```
 let new_total_rewards = newly_accumulated_rewards
  .safe_add(&reward.total_rewards.into())
  .map_err(|_| RewardAccumulationCalculationError::ArithmeticError)?;
```
I think this will never result in an error, as it should be accumulating `u128::MAX` for this to happen. However, I don't know what values can `reward.reward_rate.amount` take in order to establish some practical boundaries and be sure that `newly_accumulated_rewards + reward.total_rewards` will never overflow.

Reply:
makes sense, we can review the logic there at a later date :+1: